front/app/hooks/files/useSyncFiles.ts

```diff
@@ -0,0 +1,151 @@
+import { useCallback } from 'react';
+
+import { isString } from 'lodash-es';
+
+import useAddPhaseFile from 'api/phase_files/useAddPhaseFile';
+import useDeletePhaseFile from 'api/phase_files/useDeletePhaseFile';
+import useUpdatePhaseFile from 'api/phase_files/useUpdatePhaseFile';
+import useAddProjectFile from 'api/project_files/useAddProjectFile';
+import useDeleteProjectFile from 'api/project_files/useDeleteProjectFile';
+import useUpdateProjectFile from 'api/project_files/useUpdateProjectFile';
+
+import { SyncFilesArguments } from './types';
+
+export function useSyncFiles() {
+  // ***** Project File Hooks *****
+  const { mutateAsync: addProjectFile } = useAddProjectFile();
+  const { mutateAsync: updateProjectFile } = useUpdateProjectFile();
+  const { mutateAsync: deleteProjectFile } = useDeleteProjectFile();
+
```

Comment:
I would split this up in `useSyncProjectFiles` and `useSyncPhaseFiles`, since there is no shared logic between these two use cases in this hook